### PR TITLE
Structure

### PR DESCRIPTION
**Pull request by Thomas-dot-G**
#39 struct xml

### CODE REVIEW

File: src/model/Shape.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package model;
+
+
+public abstract class Shape implements ShapesProperties {
+	
+	/************************************
```
Comment (anqitong) [inline src/model/Shape.java:6]: Shape is an abstract class. Why is it also an interface according to the class diagramm picture?

File: src/model/specific_path/Text.java [hunk 2]
```diff
@@ -1,9 +1,9 @@
 package model.specific_path;
 
 import model.Pencil;
-import model.ShapesType;
+import model.Shape;
 
-public class Text extends ShapesType {
+public class Text extends Shape {
```
Comment (anqitong) [inline src/model/specific_path/Text.java:6]: If I understand well, ShapesType is now renamed to Shape right?

File: src/xml/EllipseXML.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package xml;
+
+import view.PencilState;
+import model.specific_path.Ellipse;
+
+public class EllipseXML extends ShapeXML {
+
```
Comment (anqitong) [inline src/xml/EllipseXML.java:9]: What if we also create a ShapeXMLFactory that creates EllipseXML, DrawingXML, ...etc?

File: src/model/Shape.java (hunk 1 above)
Comment (Thomas-dot-G) [inline src/model/Shape.java:6]: Because now in Java8, interface, abstract classes are alike.
It is in theory an interface, but as some code can be combined (color), it is better I think
For now, the color is simple, but after we can add gradient color, transparency...

File: src/model/specific_path/Text.java (hunk 2 above)
Comment (Thomas-dot-G) [inline src/model/specific_path/Text.java:6]: Exactly

File: src/xml/EllipseXML.java (hunk 3 above)
Comment (Thomas-dot-G) [inline src/xml/EllipseXML.java:9]: Yes, I haven't done that yet, but I guess we should review the Factories.
Should I do that here or in another PR ?

Comment (anqitong) [inline src/xml/EllipseXML.java:9]: It is just an idea, let's add it to issues to better think about it later

File: src/view/PencilState.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package view;
+
+public interface PencilState {
+    /**
+     * Render the pencil aspect
+     * @param pencil
```
Comment (starsasumi) [inline src/view/PencilState.java]: There is a small warning in javadoc, method no longer has param pencil.

**Comment by starsasumi** [comment]
I really like this PR, all work like a charm! Thx Thomas. Meanwhile, I'm thinking about the purpose of keeping ShapesProperties instead of using one Shape interface. (Maybe I have not caught up all conversation)

File: src/xml/EllipseXML.java (hunk 3 above)
Comment (starsasumi) [inline src/xml/EllipseXML.java:9]: We already "reserve a position" for the factory in `Presentation` interface, it's the method `ShapeState getShapeState(Shape shape);`

**Comment by Thomas-dot-G** [comment]
Yes, I was thinking the same thing. I didn't wanted to change everything, but if that works (after resolving the conflicts) we can adjust the structure in another PR, and removed what is no more necessary (some abstract methods, or interface).
As do we put interfaces with code or keep abstract classes? @starsasumi

File: src/xml/EllipseXML.java (hunk 3 above)
Comment (Thomas-dot-G) [inline src/xml/EllipseXML.java:9]: Oh, that is the actual use of that function, I didn't know what is was for^^
I understand, but actually that is not really a fabric. I mean, for the user he will still have to create the xmlShape, then set the actual shape.
The other way around would be nice. In the presentation (ex drawing, the user addSpapes, and the drawing creates the shapesXML

Comment (anqitong) [inline src/xml/EllipseXML.java:9]: That was also what I meant. Creating ShapesXML in a way that the user doesn't have to it himself

**Comment by anqitong** [comment]
The code seems fine to me.

**Comment by starsasumi** [comment]
I prefer interface with codes, it seems cooler. But it's really not that different. Depends on you @Thomas-dot-G . And I am ok to merge now.

**Comment by Thomas-dot-G** [comment]
Ok, as you wish.
I really don't mind.
For now I merge, will see in another PR then.
